ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/P2TargetUtils.java

```diff
@@ -1090,7 +1124,7 @@ private void resolveWithPlanner(ITargetDefinition target, IProfile profile, IPro
 		List<IArtifactRepository> extraArtifactRepositories = new ArrayList<>();
 		List<IMetadataRepository> extraMetadataRepositories = new ArrayList<>();
 		addAdditionalProvisionIUs(target, extraArtifactRepositories, extraMetadataRepositories);
-		ProvisioningContext context = new ProvisioningContext(getAgent()) {
+		ProvisioningContext context = new ProvisioningContext(getAgent(), getIncludeReferences()) {
```

Comment:
As mentioned on the P2 PR I don't think we need an additional constructor parameter here but can just set the property like below.